hcl_writer.go

```diff
@@ -472,6 +474,13 @@ func (w *ObjectWalker) convertCtyValue(val interface{}) cty.Value {
 	case int:
 		return cty.NumberIntVal(int64(val.(int)))
 	case int32:
+		// On volume source blocks, the mode and default_mode attributes are now mandatorily a string representation of an octal value with a leading zero
```

Comment:
I did not find any similar exceptions, this looks like the first case such a conversion is mandated.
While somewhat _hacky_, this seemed to me the most adequate place to implement this right now.
This may not scale well if others are needed in the future.

Reply:
Yes, time will tell if there are going to be a lot of these special cases. For now I think this is as good a place as any.